ui/src/alerting/actions/checks.ts

```diff
@@ -140,50 +141,59 @@ export const saveCheckFromTimeMachine = () => async (
   dispatch: Dispatch<any>,
   getState: GetState
 ) => {
-  const state = getState()
-  const {
-    orgs: {
-      org: {id: orgID},
-    },
-  } = state
+  try {
+    const state = getState()
+    const {
+      orgs: {
+        org: {id: orgID},
+      },
+    } = state
 
-  const {
-    draftQueries,
-    alerting: {check},
-  } = getActiveTimeMachine(state)
-
-  const labels = get(check, 'labels', []) as Label[]
-  const checkWithOrg = {
-    ...check,
-    query: draftQueries[0],
-    orgID,
-    labels: labels.map(l => l.id),
-  } as PostCheck
-
-  const resp = check.id
-    ? await api.patchCheck({checkID: check.id, data: checkWithOrg})
-    : await api.postCheck({data: checkWithOrg})
-
-  if (resp.status === 201 || resp.status === 200) {
-    dispatch(setCheck(resp.data))
-    dispatch(checkChecksLimits())
-  } else {
-    throw new Error(resp.data.message)
+    const {
+      draftQueries,
+      alerting: {check},
+    } = getActiveTimeMachine(state)
+
+    const labels = get(check, 'labels', []) as Label[]
+
+    const checkWithOrg = {
+      ...check,
+      query: draftQueries[0],
+      orgID,
+      labels: labels.map(l => l.id),
+    } as PostCheck
+
+    const resp = check.id
+      ? await updateCheckFromTimeMachine(checkWithOrg)
+      : await api.postCheck({data: checkWithOrg})
+
+    if (resp.status === 200 || resp.status === 201) {
+      dispatch(setCheck(resp.data))
+      dispatch(checkChecksLimits())
+
+      dispatch(push(`/orgs/${orgID}/alerting`))
+      dispatch(setTimeMachineCheck(RemoteDataState.NotStarted, null))
+    } else {
+      throw new Error(resp.data.message)
+    }
+  } catch (e) {
+    console.error(e)
```

Comment:
I think no in this case- since these are very reasonable user errors that we are alerting the user of- like they are trying to create a check with a name duplication etc. It is not a broken state in the UI. Thanks for thinking of it and reminding me though I'm sure there are places I should be sending honeybadger errors.